Add ability to configure gem, Refactoring

danielvysotskyi opened the pull request:
- Refactoring
   - change structure
   - make the structure more separated.
- Change the name of the gem to `LocalizeRubyClient`
- Add support for `Railties` for automatic rake tasks load into a `RoR` project after the gem install
- Add the ability to provide configurations for the gem.
- Change rake tasks
- Change the Readme file
  - add info about available rake tasks
  - add info about Railties support
- Add more tests

danielvysotskyi committed:
add ability to configure gem, refactoring

danielvysotskyi committed:
New tests, rubocop lint fixes

danielvysotskyi committed:
Added test for rake tasks, polishing change log, and readme

garciaf commented on lib/client/jwt_helper.rb:
```diff
@@ -0,0 +1,20 @@
+module Client
+  # The JwtHelper module provides a method to generate a JWT token.
+  # The token is signed using the app's id and includes an expiration time.
+  module JwtHelper
```
This does not need to be a separate module since you are using it, in one single place only, Please merge to the client

garciaf commented on lib/client/railtie.rb:
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Client
+  # This class integrates the gem with a Rails application if Rails is present.
```
I did not know about it, good finding

garciaf commented on lib/localize_ruby_client.rb:
```diff
@@ -0,0 +1,76 @@
+# providing methods to update translations, request translations, and upload files.
+class LocalizeRubyClient
+  include HTTParty
+  include Client::JwtHelper
+  include Client::ArchiveHelper
```
This method can be added directly in the class I do not see the value of having them out, especially if they are only used here, and the class is less than 100 lines

danielvysotskyi commented on spec/client/config_spec.rb:
```diff
@@ -0,0 +1,59 @@
+          config.validate!
+        end.to raise_error(NameError,
+                           "Error: uninitialized constant APP_ID in .env, Error: uninitialized constant PRIVATE_KEY in"\
+                           " .env, Error: uninitialized constant ROOT_PATH_TO_SAVE in .env")
```
outdated errors about .env 🫠
will fix the wording to an undefined config variable.

danielvysotskyi committed:
Feedback from Fabien, fix legacy error messages

danielvysotskyi committed:
bump version

danielvysotskyi committed:
Completely remove dotenv support, fix rake tasks path resolving, set active support as runtime dependency

danielvysotskyi commented:
Test RoR target project has successfully autoloaded rake tasks

`rake -T`
<img width="1439" alt="Screenshot 2024-07-03 at 16 58 39" src="https://github.com/test-IO/localize-ruby-client/assets/66424607/9d508cdf-c23d-4f6d-b67b-b3c5e43a77bc">

danielvysotskyi committed:
make sure that Rails env is loaded before rake task invocation

garciaf commented on README.md:
```diff
@@ -1,32 +1,38 @@
-PROJECT_UID=your_project_id_here
-ROOT_PATH_TO_SAVE=path_you_want_to_save_translated
-# usually last one is ./config/locales/
+LocalizeRubyClient.configure do |config|
```
Very very good!

danielvysotskyi committed:
feedback from Fabien
- change Readme
- make gem Rails specific
- change tests
- add rails app mock for tests

garciaf commented on lib/client/config.rb:
```diff
@@ -0,0 +1,34 @@
+      @private_key = nil
+      @project_uid = nil
+      @locales_dir_path = Rails.root.join("config", "locales").to_s
+      @site = "http://localhost:6002/api"
```
I would suggest to put the production site API by default 

```suggestion
      @site = "https://localize.cirro.io/api"
```

danielvysotskyi replied:
Yes 😅

garciaf reviewed (approved):
Make sure the config point to production directly it should be as easy as possible to install and run on any project

danielvysotskyi committed:
fix development leftovers

danielvysotskyi committed:
do not load rake tasks by default

danielvysotskyi committed:
fix README typo

danielvysotskyi committed:
store files directly in `config/locales` directory without subdirectories

danielvysotskyi committed:
Update CHANGELOG.md
